frontend/src/app/testQueue/QueueItem.tsx

```diff
@@ -316,7 +329,11 @@ const QueueItem = ({
           deviceId,
           dateTested,
           specimenTypeId: specimenId,
-          results: cacheTestResults,
+          results: doesDeviceSupportMultiPlex(deviceId)
+            ? cacheTestResults
+            : cacheTestResults.filter(
```

Comment:
this is the fix for the issue on the frontend, if the device doesnt support multiplex, then we should clear the cached results from non-covid results.

Reply:
> I confirmed that the logic to remove the results when the device is switched works awesome and the result rows for the diseases that are no longer in the test card are removed correctly. One thing that I am not able to understand is why the decision to keep different records for the result data for the test event? If upon submission the rows in the result table will not be able to be changed anymore and they are "settled" why do they need to be replicated to represent the same information for the test event?

When correcting a result, we reopen the TestOrder, then we can rewrite the results for the testorder. If click (x) on the test queue, then that should not alter the result of the testEvent.

The new system locks the Results with the testEvent.
To be honest, the correction system was broken when multiplex was added. 

> Also will there be a ticket to remove the result column from the test_order and the test_event table?. It took me some checking to realize these columns are not use anymore so cleaning the schema would be beneficial to avoid future confusion.

I agree, this is due to a lack of cleanup after the multiplex feature was "released".